### PR TITLE
Reduce benchmark suite running time

### PR DESCRIPTION
**Pull request by rayrapetyan**
### What was wrong?
https://github.com/ethereum/py-evm/issues/1607

### How was it fixed?
Reduced benchmark suite running time

### CODE REVIEW

File: scripts/benchmark/checks/simple_value_transfers.py [hunk 1]
```diff
@@ -94,15 +83,14 @@ def execute(self) -> DefaultStat:
 
         return total_stat
 
-    def mine_blocks(self, chain: MiningChain, num_blocks: int) -> Tuple[int, int]:
+    def mine_blocks(self, chain: MiningChain, num_blocks: int, num_tx: int) -> Tuple[int, int]:
         total_gas_used = 0
         total_num_tx = 0
 
         for i in range(1, num_blocks + 1):
-            num_tx = chain.get_block().header.gas_limit // SIMPLE_VALUE_TRANSFER_GAS_COST
```
Comment (carver) [inline scripts/benchmark/checks/simple_value_transfers.py]: Hm, I see where you're going, but this hard-coding approach for the number of transactions seems problematic. It requires the configuration to be precisely aware of the gas block limit in order to fill it up. Maybe the configuration parameter could be the percentage of the block to fill up? (defaulting to 1.0)

File: scripts/benchmark/checks/deploy_dos.py [hunk 2]
```diff
@@ -227,6 +230,8 @@ def create_empty_contract_revert(self, chain: MiningChain) -> None:
 
 
 class DOSContractDeployBenchmark(BaseDOSContractBenchmark):
+    def __init__(self, num_blocks: int = DEFAULT_NUM_BLOCKS, num_tx: int = DEFAULT_NUM_TX) -> None:
+        super().__init__(num_blocks, num_tx)
```
[the remark this answers is not in the document]
Comment (rayrapetyan) [reply scripts/benchmark/checks/deploy_dos.py]: @carver, oh wow... In Python derived class inherits base class "constructor", good to know!

File: scripts/benchmark/checks/simple_value_transfers.py [hunk 3]
```diff
@@ -94,15 +83,14 @@ def execute(self) -> DefaultStat:
...
             block = self.mine_block(chain, i, num_tx)
-            total_num_tx = total_num_tx + len(block.transactions)
             total_gas_used = total_gas_used + block.header.gas_used
+            total_num_tx = total_num_tx + len(block.transactions)
```
[the remark this answers is not in the document]
Comment (rayrapetyan) [reply scripts/benchmark/checks/simple_value_transfers.py]: @carver, this caused the longest delay in this benchmark.

[the remark this answers is not in the document]
Comment (rayrapetyan) [reply scripts/benchmark/checks/simple_value_transfers.py]: @carver, to keep similarity with mine_blocks from erc20_interact and deploy_dos.

File: scripts/benchmark/checks/simple_value_transfers.py (hunk 1 above)
Comment (rayrapetyan) [reply scripts/benchmark/checks/simple_value_transfers.py]: @carver, do we really need to fill it up completely? Why don't we care about same in erc20_interact and deploy_dos?

File: scripts/benchmark/checks/deploy_dos.py [hunk 4]
```diff
@@ -53,10 +53,13 @@
 
 W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
 
+NUM_BLOCKS = 20
+NUM_TX = 10
```
Comment (carver) [inline scripts/benchmark/checks/deploy_dos.py]: Heh, so these ones *are* default values, and I think the `DEFAULT_` prefix makes them clearer.
```suggestion
DEFAULT_NUM_TX = 10
```

File: scripts/benchmark/checks/simple_value_transfers.py (hunk 1 above)
Comment (carver) [reply scripts/benchmark/checks/simple_value_transfers.py]: Yup, totally fair point. This is a fundamental change to how the benchmark works, but it's a change *toward* consistency with the other benchmarks. :+1: no change needed.

File: scripts/benchmark/run.py [hunk 5]
```diff
@@ -48,6 +50,24 @@
     bold_red,
 )
 
+NUM_BLOCKS_TO_BENCHMARK = 2
+NUM_TX_TO_BENCHMARK = 1
```
Comment (carver) [inline scripts/benchmark/run.py]: Maybe use `2` transactions here, just to make sure that there are no crashes while looping over the transactions.

```suggestion
NUM_TX_TO_BENCHMARK = 2
```

File: scripts/benchmark/checks/erc20_interact.py [hunk 6]
```diff
@@ -57,10 +57,12 @@
 
 W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
 
+NUM_BLOCKS = 100
```
Comment (carver) [inline scripts/benchmark/checks/erc20_interact.py]:
```suggestion
DEFAULT_NUM_BLOCKS = 100
```

File: scripts/benchmark/checks/erc20_interact.py [hunk 7]
```diff
@@ -57,10 +57,12 @@
...
+NUM_TX = 2
```
Comment (carver) [inline scripts/benchmark/checks/erc20_interact.py]:
```suggestion
DEFAULT_NUM_TX = 2
```

Comment (carver) [inline scripts/benchmark/checks/erc20_interact.py]:
```suggestion
DEFAULT_NUM_TX = 2
```

File: scripts/benchmark/checks/deploy_dos.py [hunk 8]
```diff
@@ -53,10 +53,13 @@
 
 W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
 
+NUM_BLOCKS = 20
```
Comment (carver) [inline scripts/benchmark/checks/deploy_dos.py]:
```suggestion
DEFAULT_NUM_BLOCKS = 20
```

**Comment by rayrapetyan** [comment]
@carver, is there anything awaiting any changes from me? I think this patch should significantly decrease tests time so earlier we merge it is better for devs here. Thanks.

**Comment by pipermerriam** [comment]
@rayrapetyan looks like @carver left a number of suggested changes.  If you can review/implement/respond to those I think we can move this forward.

**Comment by carver** [comment]
Let's reopen this later, if it becomes necessary again. It's not a huge change, the PR has been quiet a long time, and benchmarking has been fast enough lately.

**Pull request closed without merging**